pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -63,8 +63,8 @@ func New(mounterPath string) *Mounter {
 
 func (m *Mounter) Mount(ctx context.Context, mc *MountConfig) error {
 	// Start the token server for HostNetwork enabled pods.
-	if mc.TokenServerIdentityProvider != "" {
-		klog.Infof("Pod has hostNetwork enabled and token server feature is turned on. Starting Token Server on %s/%s", mc.TempDir, TokenFileName)
+	if mc.TokenServerIdentityProvider != "" && mc.HostNetworkKSAOptIn {
```

Comment:
for the scenario where user does not explicitly provide the pod KSA config, we still need to let mount succeed, else we will break such customers who are already using node SA + pod host network